bugsnag-android-core/src/main/java/com/bugsnag/android/StateEvent.kt

```diff
@@ -0,0 +1,43 @@
+package com.bugsnag.android
+
+sealed class StateEvent {
+    class Install(
+        val autoDetectNdkCrashes: Boolean,
+        val appVersion: String?,
+        val buildUuid: String?,
+        val releaseStage: String?
+    ) : StateEvent()
+
+    object DeliverPending : StateEvent()
```

Comment:
I'm sure there is a good reason for this, but coud you explain why the entries that don't need to modify the signature of the constructor/instance variables have the keyword `object` and not `class`?

Reply:
`object` defines a singleton class in Kotlin. As `DeliverPending` doesn't have any fields that can alter value, declaring a singleton means we avoid having to instantiate a new `DeliverPending` object each time.